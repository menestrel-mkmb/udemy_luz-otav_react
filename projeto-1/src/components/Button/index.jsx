import { Component } from "react";

class Button extends Component {
    render() {
        const { onClick, text } = this.props;
        return (
            <button
                className="button"
                onClick={onClick}
            >
                {text}
            </button>
        );
    }
}

export default Button;