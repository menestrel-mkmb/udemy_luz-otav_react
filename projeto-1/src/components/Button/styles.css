button {
    background-color: #fff;
    border: 0;
    border-radius: 0.25rem;
    box-shadow: 0 0 10px rgba( 0, 0, 0, .2 );

    width: 100%;

    margin: 1rem 2rem;
    padding: .5rem 1rem;
}