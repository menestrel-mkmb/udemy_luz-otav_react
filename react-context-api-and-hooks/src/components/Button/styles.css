.counter-btn {
  width: 80px;
}
