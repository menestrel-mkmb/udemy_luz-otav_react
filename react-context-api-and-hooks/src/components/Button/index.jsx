import P from 'prop-types';
import './styles.css';

export const Button = ({ children, onClick }) => {
  return (
    <button className="counter-btn" onClick={() => onClick()}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: P.node.isRequired,
  onClick: P.func.isRequired,
};
