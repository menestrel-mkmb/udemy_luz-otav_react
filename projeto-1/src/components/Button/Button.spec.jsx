import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Button from '.';

describe('<Button /> See more posts', () => {
    it('should render the button with the text "Load more posts"', () => {
        expect.assertions(1);
        render(<Button text='Load more posts' />);

        const button = screen.getByRole('button', { name: /load more posts/i });

        expect(button).toBeInTheDocument();
    });

    it('should call fn on button click', () => {
        const fn = jest.fn();
        expect.assertions(1);
        render(<Button text='Load more posts' onClick={fn} />);

        const button = screen.getByRole('button', { name: /load more posts/i });
        userEvent.click(button);

        expect(fn).toHaveBeenCalledTimes(1);
    });

    it('should be disabled when disabled is true', () => {
        expect.assertions(1);
        render(<Button text='Load more posts' disabled={true} />);

        const button = screen.getByRole('button', { name: /load more posts/i });

        expect(button).toBeDisabled();
    });

    it('should be enabled when disabled is false', () => {
        expect.assertions(1);
        render(<Button text='Load more posts' disabled={false} />);

        const button = screen.getByRole('button', { name: /load more posts/i });

        expect(button).not.toBeDisabled();
    });

    it('should match snapshot', () => {
        const fn = jest.fn();
        const { container } = render(<Button
                                        text='Load more posts'
                                        onClick={fn}
                                        disabled={false}
                                    />);
        expect(container.firstChild).toMatchSnapshot();
    });
});