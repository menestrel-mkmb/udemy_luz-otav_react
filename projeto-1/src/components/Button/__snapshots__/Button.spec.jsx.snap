// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> See more posts should match snapshot 1`] = `
<button
  class="button"
>
  Load more posts
</button>
`;
